projects/npm-tools/packages/npm-scripts/test/sass/build.js

```diff
@@ -0,0 +1,265 @@
+/**
+ * SPDX-FileCopyrightText: © 2020 Liferay, Inc. <https://liferay.com>
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+const fs = require('fs');
+const os = require('os');
+const path = require('path');
+const rimraf = require('rimraf');
+
+const buildSass = require('../../src/sass/build');
+const log = require('../../src/utils/log');
+
+jest.mock('../../src/utils/log');
+
+const FIXTURES = path.resolve(__dirname, '../../__fixtures__/sass');
+const OUTPUT_SASS_DIR = '.sass-cache';
+
+describe('sass', () => {
+	let tempDir;
+	let tempSassDir;
+
+	beforeAll(() => {
+		tempDir = fs.mkdtempSync(path.join(os.tmpdir(), 'scripts-'));
+
+		tempSassDir = path.join(tempDir, OUTPUT_SASS_DIR);
+	});
+
+	afterEach(() => {
+		log.mockReset();
+
+		rimraf.sync(tempDir);
```

Comment:
You don't need this. One of the main reasons to use `os.tmpdir()` is that the OS takes care of cleaning things up at some opportune time (usually after a reboot, but it varies by OS; some will reclaim space on a schedule, or when space runs low, I believe) — this in turn means that the tests run faster because they do less work.

Reply:
@wincent I ended up needing to clear the directory because I didn't want built css files to hang around between tests. If we remove this, we start to run into issues where a built file already exists.

If the tmp directory was cleared between tests we wouldn't need this, but it only cleans the tmp directory after all the tests run.

Reply:
Ah sorry, I missed that you weren't using `mkdtemp` as well in a `beforeEach` instead of `beforeAll`, like you can [see here](https://github.com/liferay/liferay-frontend-projects/blob/ed8a348fe0bbad3ac8bf28ddd5074b56f9bdd64b/projects/npm-tools/packages/npm-scripts/test/scripts/format.js#L35) in this repo. Pristine set of files for each test and you can forget about cleanup and the need to use a dependency.

Reply:
Ah, that makes sense to create a temp before each test, my brain seemed to skip the idea that each tmpdir would be different if I did it beforeEach. Thanks for clarifying!